pkg/controller/utils/discovery.go

```diff
@@ -225,15 +225,26 @@ func isEKS(ctx context.Context, c kubernetes.Interface) (bool, error) {
 // differentiate between versions, which requires another approach. In this case,
 // the presence of an "rke2" configmap in kube-system namespace is used.
 func isRKE2(ctx context.Context, c kubernetes.Interface) (bool, error) {
-	cm, err := c.CoreV1().ConfigMaps("kube-system").Get(ctx, "rke2", metav1.GetOptions{})
-	if err != nil {
-		if kerrors.IsNotFound(err) {
-			return false, nil
-		}
+	foundRKE2Resource := false
+	_, err := c.CoreV1().ConfigMaps("kube-system").Get(ctx, "rke2", metav1.GetOptions{})
+	if err == nil {
+		foundRKE2Resource = true
+	} else if !kerrors.IsNotFound(err) {
+		return false, err
+	}
+
+	// In current RKE2 the above ConfigMap no longer exists, but we leave that code in place in
+	// case there are variants where it is useful.  Check also for the RKE2 DNS service - which
+	// is especially relevant because one of the main uses of the RKE2 autodetection is to set
+	// DNS config.
+	_, err = c.CoreV1().Services("kube-system").Get(ctx, "rke2-coredns-rke2-coredns", metav1.GetOptions{})
```

Comment:
Is that right that `rke2-coredns` is duplicated? That looks odd but I could believe it.

Reply:
Here are services from an RKE2 cluster in one of our e2e runs:
```
NAMESPACE              NAME                                TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)                      AGE   SELECTOR
calico-system          calico-kube-controllers-metrics     ClusterIP   None            <none>        9094/TCP                     46m   k8s-app=calico-kube-controllers
calico-system          calico-node-metrics                 ClusterIP   None            <none>        9081/TCP,9900/TCP            47m   k8s-app=calico-node
calico-system          calico-typha                        ClusterIP   10.43.212.41    <none>        5473/TCP                     47m   k8s-app=calico-typha
default                kubernetes                          ClusterIP   10.43.0.1       <none>        443/TCP                      50m   <none>
kube-system            rke2-coredns-rke2-coredns           ClusterIP   10.43.0.10      <none>        53/UDP,53/TCP                49m   app.kubernetes.io/instance=rke2-coredns,app.kubernetes.io/name=rke2-coredns,k8s-app=kube-dns
kube-system            rke2-metrics-server                 ClusterIP   10.43.175.72    <none>        443/TCP                      47m   app=rke2-metrics-server,release=rke2-metrics-server
kube-system            rke2-snapshot-validation-webhook    ClusterIP   10.43.128.153   <none>        443/TCP                      46m   app.kubernetes.io/instance=rke2-snapshot-validation-webhook,app.kubernetes.io/name=rke2-snapshot-validation-webhook
staged-enforced-5530   svc-server-pod-zhnh5                ClusterIP   10.43.209.128   <none>        80/TCP                       73s   pod-name=server-pod-zhnh5
tigera-compliance      compliance                          ClusterIP   10.43.158.44    <none>        443/TCP                      42m   k8s-app=compliance-server
tigera-elasticsearch   tigera-elasticsearch-metrics        ClusterIP   None            <none>        9081/TCP                     42m   k8s-app=tigera-elasticsearch-metrics
tigera-elasticsearch   tigera-linseed                      ClusterIP   10.43.18.30     <none>        443/TCP                      42m   k8s-app=tigera-linseed
tigera-elasticsearch   tigera-secure-es-7f5dee596fef130f   ClusterIP   None            <none>        9200/TCP                     45m   common.k8s.elastic.co/type=elasticsearch,elasticsearch.k8s.elastic.co/cluster-name=tigera-secure,elasticsearch.k8s.elastic.co/statefulset-name=tigera-secure-es-7f5dee596fef130f
tigera-elasticsearch   tigera-secure-es-gateway-http       ClusterIP   10.43.25.99     <none>        9200/TCP,5601/TCP            43m   k8s-app=tigera-secure-es-gateway
tigera-elasticsearch   tigera-secure-es-http               ClusterIP   10.43.103.173   <none>        9200/TCP                     45m   common.k8s.elastic.co/type=elasticsearch,elasticsearch.k8s.elastic.co/cluster-name=tigera-secure
tigera-elasticsearch   tigera-secure-es-internal-http      ClusterIP   10.43.5.40      <none>        9200/TCP                     45m   common.k8s.elastic.co/type=elasticsearch,elasticsearch.k8s.elastic.co/cluster-name=tigera-secure
tigera-elasticsearch   tigera-secure-es-transport          ClusterIP   None            <none>        9300/TCP                     45m   common.k8s.elastic.co/type=elasticsearch,elasticsearch.k8s.elastic.co/cluster-name=tigera-secure
tigera-fluentd         fluentd-metrics                     ClusterIP   None            <none>        9081/TCP                     45m   k8s-app=fluentd-node
tigera-kibana          tigera-secure-kb-http               ClusterIP   10.43.136.162   <none>        5601/TCP                     45m   common.k8s.elastic.co/type=kibana,kibana.k8s.elastic.co/name=tigera-secure
tigera-manager         tigera-manager                      ClusterIP   10.43.202.144   <none>        9443/TCP                     42m   k8s-app=tigera-manager
tigera-packetcapture   tigera-packetcapture                ClusterIP   10.43.214.228   <none>        443/TCP                      46m   k8s-app=tigera-packetcapture
tigera-prometheus      alertmanager-operated               ClusterIP   None            <none>        9093/TCP,9094/TCP,9094/UDP   45m   app.kubernetes.io/name=alertmanager
tigera-prometheus      calico-node-alertmanager            ClusterIP   10.43.229.145   <none>        9093/TCP                     46m   alertmanager=calico-node-alertmanager
tigera-prometheus      prometheus-http-api                 ClusterIP   10.43.108.180   <none>        9090/TCP                     46m   prometheus=calico-node-prometheus
tigera-prometheus      prometheus-operated                 ClusterIP   None            <none>        9090/TCP                     45m   app.kubernetes.io/name=prometheus
tigera-system          tigera-api                          ClusterIP   10.43.189.184   <none>        443/TCP,8080/TCP             46m   apiserver=true
```